File: 19-03-24-done/Q2.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        if root == None:
            return []
        que = [root]
        res = []
        while que:
            res_t = []
            for _ in range(len(que)):
                temp = que.pop(0)
                res_t.append(temp.val)
                if temp.left != None:
                    que.append(temp.left)
                if temp.right !=None:
                    que.append(temp.right)
            res.append(res_t)
        return ress

File: 19-03-24/q2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    void exploreTree(vector<vector<int> >& data, int depth, TreeNode* ptr){
        if(ptr == nullptr)
            return;
        if(depth<data.size())
            data[depth].emplace_back(ptr->val);
        else{
            vector<int> tmp = {ptr->val};
            data.emplace_back(tmp);
        }
        exploreTree(data, ++depth, ptr->left);
        exploreTree(data, ++depth, ptr->right);
    }
    
    vector<vector<int>> levelOrder(TreeNode* root) {
        int depth = 0;
        vector<vector<int> > result;
        exploreTree(result, depth, root);
        return result;
    }
};
